src/main/scala/com/azavea/landsatutil/Landsat8Query.scala

```
@@ -113,56 +123,69 @@ class Landsat8Query() {
-    val url = s"${Landsat8Query.API_URL}?$search"
-    val system = ActorSystem(s"landsat_query_collect_${java.util.UUID.randomUUID}")
-    try {
-      val count = HttpClient.get[QueryResult](url, system).metadata.total
+    val termSearch = s"search=$searchTerms"
 
-      val groups = {
-        val g = count / 100
-        if(g * 100 == count) g else g + 1
-      }
+    def numGroups(total: Int): Int = {
+      val g = total / 100
+      if (g * 100 == total) g else g + 1
+    }
 
-      (0 until groups)
-        .map { g =>
-          val skip = g * 100
-          s"${Landsat8Query.API_URL}?$search&limit=100&skip=$skip"
-        }
-        .par
-        .flatMap { url => HttpClient.get[QueryResult](url, system).images }
-        .filter(_filterFunction)
-        .toList
-    } catch {
-      case e: spray.httpx.UnsuccessfulResponseException if e.response.status.intValue == 404 =>
-        // Not found
-        Seq()
-    } finally {
-      system.shutdown()
+    def queryGroup(groupNum: Int) = {
+      val skip = groupNum * 100
+      Uri.Query(s"$termSearch&limit=100&skip=$skip")
     }
 
+    val client = new HttpClient(Landsat8Query.API_URL)
+    // Imports execution context for `Future`s to run in.
+    import client.system.dispatcher
+
+    Try {
+      try {
+        // TODO: Make use of Host-level connection pool
+        val parallelRequests =
+          client.get[QueryResult](Uri.Query(termSearch))
+            .flatMap(query ⇒ {
+              val groups = numGroups(query.metadata.found)
+              val groupImages =
+                for (g ← 0 until groups) yield {
+                  client
+                    .get[QueryResult](queryGroup(g))
+                    .map(_.images.filter(_filterFunction))
+                }
+              Future.sequence(groupImages).map(_.flatten)
+            })
+
+        // Fork-join pattern...
+        Await.result(parallelRequests, timeout)
+      } finally {
+        client.shutdown()
+      }
+    }
```

@metasim Uses/returns `Try` to avoid swallowing exceptions.